### PR TITLE
Support mtl-2.3-rc3

### PR DESCRIPTION
**Pull request by Bodigrim**
See https://discourse.haskell.org/t/ann-release-candidate-for-mtl-2-3/3687

### CODE REVIEW

File: src/System/Random/Stateful.hs [hunk 1]
```diff
@@ -105,10 +105,11 @@ module System.Random.Stateful
   ) where
 
 import Control.DeepSeq
+import Control.Monad (replicateM)
 import Control.Monad.IO.Class
 import Control.Monad.ST
 import GHC.Conc.Sync (STM, TVar, newTVar, newTVarIO, readTVar, writeTVar)
-import Control.Monad.State.Strict
+import Control.Monad.State.Strict (MonadState, state)
```
Comment (lehins) [inline src/System/Random/Stateful.hs:112]: I can't see why these changes are needed, can you revert them?

Comment (Bodigrim) [reply src/System/Random/Stateful.hs:112]: They are needed. [Changelog](https://hackage.haskell.org/package/mtl-2.3/candidate/changelog) says "Remove re-exports of `Control.Monad`, `Control.Monad.Fix` and `Data.Monoid` modules".

Comment (lehins) [reply src/System/Random/Stateful.hs:112]: Gotcha. It does make sense now. Thank you.

**Comment by andreasabel** [comment]
@Bodigrim @lehins Could this fix be backported to `random-1.2.1` and be released, say as `random-1.2.1.1`?

I tried to use this PR directly as `source-repository` when testing for compatibility with `mtl-2.3-rc3` in some of my packages, but then I get `random-1.3.0` which dropped some older GHCs.  So I cannot really use it when I want to support GHC 7.8 and older.

**Comment by lehins** [comment]
@andreasabel I can certainly do that. As soon as mtl-2.3 is out I'll make such release.

**Comment by Bodigrim** [comment]
@andreasabel given amount and magnitude of last-minute changes in `mtl-2.3`, I advise against speculative releases of downstream libraries.

**Comment by andreasabel** [comment]
> @andreasabel given amount and magnitude of last-minute changes in `mtl-2.3`, I advise against speculative releases of downstream libraries.

Fair enough!

**Comment by sjakobi** [comment]
mtl-2.3 has been released. Could this patch be released too? :)

**Comment by lehins** [comment]
`random-1.2.1.1` released last night :+1:

**Pull request merged**